biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/GenbankReader.java

```diff
@@ -153,35 +154,56 @@ public GenbankReader(
 		}
 
 		LinkedHashMap<String,S> sequences = new LinkedHashMap<>();
-		@SuppressWarnings("unchecked")
 		int i=0;
 		while(true) {
 			if(max>0 && i>=max) break;
 			i++;
-			String seqString = genbankParser.getSequence(bufferedReader, 0);
-			//reached end of file?
-			if(seqString==null) break;
-			@SuppressWarnings("unchecked")
-			S sequence = (S) sequenceCreator.getSequence(seqString, 0);
-			genbankParser.getSequenceHeaderParser().parseHeader(genbankParser.getHeader(), sequence);
-
-			// add features to new sequence
-			genbankParser.getFeatures().values().stream()
-			.flatMap(List::stream)
-			.forEach(sequence::addFeature);
-
-			// add taxonomy ID to new sequence
-			ArrayList<DBReferenceInfo> dbQualifier = genbankParser.getDatabaseReferences().get("db_xref");
-			if (dbQualifier != null){
-				DBReferenceInfo q = dbQualifier.get(0);
-				sequence.setTaxonomy(new TaxonomyID(q.getDatabase()+":"+q.getId(), DataSource.GENBANK));
+			S sequence = getSequence();
+			if(null == sequence) {
+				break;
 			}
-
 			sequences.put(sequence.getAccession().getID(), sequence);
 		}
 
 		return sequences;
 	}
+	
+	public Stream<S> getSequencesAsStream() {
+		return Stream.generate(() -> {
+			try {
+				return getSequence();
+			} catch (IOException | CompoundNotFoundException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+				return null;
```

Comment:
I'd rather throw the exception forward and let the caller handle it. Returning null means all callers have to handle the null and that they lose the ability to provide a good error message.